### PR TITLE
Pass 1 (Items 3–4) + Pass 2 (Items 5–6): Rollover, Core Tests, Refactor, Accessibility/Motion/Perf

### PR DESCRIPTION
**Pull request by KenWidemon**
# ✅ Pass 1 (Items 3–4) + Pass 2 (Items 5–6): Rollover, Core Tests, Refactor, Accessibility/Motion/Perf

## Summary
This PR implements baseline **day rollover behavior** (refresh daily content on app foreground when the local day changes), adds **core unit tests** to lock down decoding and deterministic selection, then follows with a **post-tests refactor** of the daily view for maintainability. It also includes a combined **Accessibility + Motion + Performance** pass to improve production readiness without changing core product behavior.

---

### Pass 1 — Day Rollover Behavior (Baseline)
- Implemented foreground-based refresh:
  - When the app returns to foreground, detect whether the **local day changed**.
  - If changed, refresh the daily content entry.
- Baseline explicitly **does not** include a midnight timer refresh (deferred as optional enhancement).

### Pass 1 — Tests for Content + Selection + Rollover
- Added unit tests for:
  - JSON manifest/schema decoding
  - Deterministic daily selection: `YYYYMMDD % count`
  - Fallback handling (e.g., missing/invalid content scenarios as implemented)
  - Rollover expectations (foreground refresh triggers only when day changes)

### Pass 2 — Refactor Daily View (Post-Tests)
- Split large `DailyWhiskersView` logic into focused components and/or lightweight helpers
- Preserved behavior and visuals while reducing file size and complexity
- Improved separation of concerns to make future UI tuning safer/faster

### Pass 2 — Accessibility + Motion + Performance
- Accessibility improvements:
  - Dynamic Type support (text scaling without layout breakage)
  - VoiceOver labels/hints for key UI elements
  - Contrast adjustments where needed
- Motion support:
  - Reduced motion fallback behavior for sparkles/animations
- Lightweight performance cleanup:
  - Tuned sparkle/canvas animation cost (where applicable)
  - Avoided unnecessary redraw work and overdraw in core views

---

## Why
- **Predictability:** Users should reliably see new daily content after the day changes.
- **Correctness:** Deterministic content selection and schema decoding must be protected by tests.
- **Maintainability:** Refactoring after tests prevents regressions and enables safer iteration.
- **Production readiness:** Accessibility and reduced motion handling are essential for quality and inclusivity.
- **Performance:** Animations should remain smooth across device tiers.

---

## Screenshots / Video
- [ ] Optional: short recording showing daily card refresh after background → foreground across a simulated day change
- [ ] Optional: Reduced Motion enabled preview (sparkles/animations reduced or disabled)
- [ ] Optional: Dynamic Type (Large/Accessibility sizes) screenshots for key screens

---

### Day Rollover Behavior
- [ ] Launch app; confirm daily content renders normally.
- [ ] Background the app.
- [ ] Simulate a day change (simulator date/time forward) **or** use a debug/injected date provider if present.
- [ ] Foreground the app.
- [ ] Verify the daily content refreshes when local day changed.
- [ ] Verify it does **not** refresh if the day did not change (no unnecessary churn).

### Tests
- [ ] Run unit tests locally (Cmd+U)
- [ ] Confirm new suites pass:
  - Manifest decoding
  - Deterministic selection (`YYYYMMDD % count`)
  - Fallback handling
  - Foreground rollover expectations

### Refactor Safety
- [ ] Confirm visuals and behavior match pre-refactor output:
  - Daily card composition, text placement, vibe pill, theming
  - No navigation/tooling regressions

### Accessibility + Motion + Performance
- [ ] Enable VoiceOver: verify labels/hints read correctly and focus order is sane
- [ ] Increase Dynamic Type: verify quote, vibe pill, and primary UI remain readable and not clipped
- [ ] Check contrast in at least one dark and one lighter archetype theme
- [ ] Enable Reduced Motion: verify animations/sparkles are reduced or disabled appropriately
- [ ] Quick perf sanity: scroll/transition interactions remain smooth; no obvious jank on simulator

---

## Risk / Impact
**Risk Level:** Medium  
This PR touches daily refresh behavior, core selection logic, view refactor, and accessibility/motion adjustments.

Potential risks:
- Timezone/local calendar edge cases (DST, locale calendars)
- Rollover refresh happening too often or not at all
- UI regressions introduced during refactor
- Accessibility changes subtly shifting layout

Mitigations:
- Local-day comparison (calendar-based) rather than raw time intervals
- Tests that encode rollover expectations and selection stability
- Refactor only after tests land; behavior preserved intentionally
- Manual QA pass with Dynamic Type + Reduced Motion toggles

---

## Out of Scope
- Midnight timer refresh (optional later)
- CI expansion beyond existing baseline (unless explicitly included)
- Content pipeline changes (schema changes beyond what’s necessary for tests)
- New features (favorites/sharing/notifications) not directly tied to rollover/tests/a11y

---

## Follow-ups
- [ ] Optional enhancement: midnight timer refresh strategy
- [ ] Expand tests to cover additional edge cases (DST boundary, leap day, etc.)
- [ ] Consider snapshot/UI tests if/when UI stabilizes
- [ ] Add instrumentation for rollover refresh events (if analytics/logging layer is added)

---

## Notes
- Deterministic selection remains: `YYYYMMDD % count` for stable rotation.
- Rollover baseline is foreground-driven to keep V1 simple and predictable.
- Reduced Motion behavior should preserve the experience while respecting user settings.

### CODE REVIEW

**Comment by KenWidemon** [comment]
@codex, review

File: DailyWhiskers/Views/DailyWhiskersView.swift [hunk 1]
```diff
@@ -38,354 +49,24 @@ struct DailyWhiskersView: View {
...
-            .background(
-                Capsule(style: .continuous)
-                    .fill(theme.pillFill)
-                    .overlay(
-                        Capsule(style: .continuous)
-                            .stroke(theme.pillStroke, lineWidth: 1)
-                    )
-                    .shadow(color: theme.glow.opacity(0.35), radius: 10, y: 6)
-            )
+        currentCard = provider.content(for: Date())
```
[the remark this answers is not in the document]
Comment (KenWidemon) [reply DailyWhiskers/Views/DailyWhiskersView.swift:68]: Will resolve this later
